iceoryx_examples/icedelivery/iox_subscriber_typed.cpp

```diff
@@ -39,20 +39,25 @@ int main()
     // initialized subscriber
     iox::popo::SubscriberOptions subscriberOptions;
     subscriberOptions.queueCapacity = 10U;
-    iox::popo::TypedSubscriber<RadarObject> typedSubscriber({"Radar", "FrontLeft", "Object"}, subscriberOptions);
-    typedSubscriber.subscribe();
+    iox::popo::TypedSubscriber<RadarObject> subscriber({"Radar", "FrontLeft", "Object"}, subscriberOptions);
+    subscriber.subscribe();
 
     // run until interrupted by Ctrl-C
     while (!killswitch)
     {
-        if (typedSubscriber.getSubscriptionState() == iox::SubscribeState::SUBSCRIBED)
+        if (subscriber.getSubscriptionState() == iox::SubscribeState::SUBSCRIBED)
         {
-            typedSubscriber.take()
-                .and_then([](iox::popo::Sample<const RadarObject>& object) {
-                    std::cout << "Got value: " << object->x << std::endl;
-                })
-                .if_empty([] { std::cout << std::endl; })
-                .or_else([](iox::popo::ChunkReceiveError) { std::cout << "Error receiving chunk." << std::endl; });
+            subscriber.take_1_0()
+                .and_then([](auto& sample) { std::cout << "Got value: " << sample->x << std::endl; })
+                .or_else([](auto& result) {
+                    // only has to be called if the alternative is of interest,
+                    // i.e. if nothing has to happen when no data is received and
+                    // a possible error alternative is not checked or_else is not needed
+                    if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                    {
+                        std::cout << "Error receiving chunk." << std::endl;
+                    }
+                });
```

Comment:
Maybe something for the future. What do you thing about a `or_else_if(E c, [](E& r))` and `or_else_if_not(E c, [](E& r))` which calls the lambda only if `c == r` or `c != r`. It might be helpful for such cases where you want to ignore some of the error cases or act only on a specific error case.

It might be dangerous though, since you could still call `or_else` afterwards and it would still be executed, which is semantically different than an `else if` statement. This could be solved if `or_else` would return a reference to itself, but a new `expected` when executed

Reply:
Oh this reminds me of the old days: `pop_if`. But in this case I would vote against it. I played around with your idea and in my opinion the code is less readable then with an `if`-statement. Take a look:
```cpp
.or_else([](auto& result) {
                    if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
                    {
                        std::cout << "Error receiving chunk." << std::endl;
                    });
```
```cpp
.or_else_if( [](auto & result) { return result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE; }
            ,[](auto& result) { std::cout << "Error receiving chunk." << std::endl; });
```

Reply:
@elfenpiff you misunderstood me. My idea was more like this
```c++
.or_else_if_not(iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE, [](auto& result) {
    std::cout << "Error receiving chunk." << std::endl; 
})
```

It's problematic that you still could use `or_else_if(...).or_else(...)` and both of them could be executed. So, a flag would be needed to remember if the `else` branch was already executed. This might still be a good idea, even without implementing `or_else_if` to catch a `.or_else().or_else()` chain.

Reply:
@elBoberido I would consider it to be too confusing. I envisioned returning a ``ReceiveResult<T, ReceiveStatus>`` which inherits from ``expected<T, ReceiveStatus>`` and hence provides ``and_then`` etc. In addition it could have ``if_empty`` (which would internally check the Status) or what have you. This is also quite extension friendly for the receive use case, as it does not need to be too general. We could also have a slightly more specific ``if_status(Status, action)`` or general form ``if(condition, action)``. 

I already proposed this but nobody wanted to listen so...

Reply:
If desired it can be added later, it will not invalidate the current usage.

Reply:
@MatthiasKillat it was just food for thought. I think with `and_then` and `or_else` currently just returning a reference it might easily lead to errors due to unintended double execution.